### PR TITLE
Like dislike

### LINKED ISSUE #106: Like & dislike
## Requirments:

-  a user can like a post
-  a user can unlike a post
-  a user can dislike a post 
-  a user can un dislike a post

### Backend:
- [x] modify the schema of Posts (if needed)
- [x] create a route for likes 
- [x] create a route for dislikes
- [x] create a route to get likes
- [x] create a route to get dislikes
- [x] controller to get likes  using postID
- [x] controller to get dislikes  using postID
- [x]  cheek if user already liked a certain post (middleware?)
- [x] controller to add a like (userId) into likes array in post schema
- [x] controller to remove a like (userId) from the likes array in post schema
- [x] controller to add a dislike (userId) into the dislikes array in post schema
- [x] controller to remove dislike (userId) from the dislikes array in post schema

### Frontend:
- [x] useLike hook
- [x] useDislike hook
- [x] build like and dislikes components
- [x] two number components to render the num of likes and dislikes
- [x] different colors for like and dislike (blue, red ? think about light and dark mode)
- [x] refetch likes array when modified
- [x] refetch dislikes array when modified

### PR DESCRIPTION
**Pull request by mohammadalnajar**
# Description

## Backend:
- routes for like and dislike post made
- routes built to get likes and dislikes
- controller made to like and dislike apost 
- middleware to check if user already like or dislike a post

## fronend:
- like & dislike component made
- getlikes and getdislikes queries made by postId
- user can now like and dislike a post
- like button with blue color (when checked)
- dislike button with red color (when checked)

Fixes #106 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project

### CODE REVIEW

File: client/src/components/post/hooks/usePostActions.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useQuery } from 'react-query';
+import { getDislikes, getLikes } from '../api';
+
+const usePostActions = ({ postId }) => {
+  const { data: { data: { likes } = {} } = {} } = useQuery(
+    [`getLikes-${postId}`, { postId }],
+    getLikes,
+    {
+      staleTime: 10 * 60 * 1000,
+      refetchInterval: 2 * 60 * 1000,
```
Comment (Hani-AS) [inline client/src/components/post/hooks/usePostActions.js:10]: Why are u refetching every 2 min when the stale time is set on 10 min? Stale time means that the data is fresh for 10min.

File: client/src/components/post/hooks/useLike.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    let check = false;
+    likes?.forEach((like) => {
+      if (like?.userId === userId) {
+        check = true;
+      }
+    });
+    return check;
+  };
+
+  const [checked, setChecked] = useState(userLikePostCheck());
```
Comment (Hani-AS) [inline client/src/components/post/hooks/useLike.js:22]: I think lazyloading the usestate function is better in this scenario. useState(() => userLikePostCheck());

File: client/src/components/post/hooks/useDislike.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    dislikes?.forEach((dislike) => {
+      if (dislike?.userId === userId) {
+        check = true;
+      }
+    });
+    return check;
+  };
+
+  const [checked, setChecked] = useState(userDislikePostCheck());
+  const toggleChecked = () => {
```
Comment (Hani-AS) [inline client/src/components/post/hooks/useDislike.js:23]: Same as the other one but do your research to understand it.

**Review by Hani-AS — COMMENTED** [review]
Looks good man. Is there no way to combine the useLikes and useDisLikes hooks or is it better like this?

File: client/src/components/post/hooks/usePostActions.js (hunk 1 above)
Comment (mohammadalnajar) [reply client/src/components/post/hooks/usePostActions.js:10]: okay then we make the staleTime also 2 mins

File: client/src/components/post/hooks/useLike.js (hunk 2 above)
Comment (mohammadalnajar) [reply client/src/components/post/hooks/useLike.js:22]: we take a look at it

**Pull request merged**